clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/server/P2PFileTransferServerHandler.java

```diff
@@ -145,6 +170,8 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest httpReque
         LOGGER.error("Failed to send all files for {}", blobTransferRequest.getFullResourceName(), future.cause());
       }
     });
+
+    blobSnapshotManager.removeConcurrentUserRestriction(isHybridStore, blobTransferRequest);
```

Comment:
this method should also be a part of `exceptionCaught` or `channelInActive`(which i think it's a better place) to decrease the concurrent user restriction, otherwise if there's some exceptions happen, the count never get decreased.